libraries/chain/include/eosio/chain/producer_schedule.hpp

```diff
@@ -78,11 +82,12 @@ namespace eosio { namespace chain {
       shared_block_signing_authority           authority;
    };
 
+   struct producer_authority_schedule;
+
    struct shared_producer_authority_schedule {
-      shared_producer_authority_schedule() = delete;
+      shared_producer_authority_schedule() = default;
 
-      explicit shared_producer_authority_schedule( chainbase::allocator<char> alloc )
-      :producers(alloc){}
+      shared_producer_authority_schedule( const producer_authority_schedule& );
```

Comment:
`explicit`

Reply:
Leaving as-is because the type of the constructor parameter leave little chance for unexpected conversions.